### PR TITLE
[math] remove implicit conversions to/from KahanSum

### PR DESCRIPTION
**Pull request by egpbos**
# This Pull request:

Follow-up on #11940; ~please merge that one before this one~ it has been merged, so this one can also be merged.

## Changes or fixes:

The auto-conversion to type `T` and implicit type `T` constructor in `KahanSum` make it hard to debug `KahanSum`, because it is easy to overlook implicit conversions in code, especially in lines where the type of the return value is `auto`. The second commit in this PR (the first commit is the same as that in #11940) removes them and where they were necessary replaces them with an explicit construction or explicit conversion to double via `Sum()`

## Checklist:

- [x] tested changes locally
- [x] updated the docs (if necessary)

### CODE REVIEW

**Comment by egpbos** [comment]
Good to hear!

Btw, now that I think about it, I think this PR doesn't actually need the first commit (which is also in #11940). It could just be ad1505066194502735875bb6fd8596eade922832. I don't have time right now to rebase the branch, but I can do it later (or someone else can do it if they feel like it).

**Comment by egpbos** [comment]
Ah, I know what the problem is there (I commented that line out in my local build, that's why it didn't fail for me). I'll fix the commit up later.

**Comment by egpbos** [comment]
Rebased on https://github.com/root-project/root/pull/11940 branch (it was more convenient to leave it based on that after all, I tried taking the commit out, but it needed too many changes) and fixed the build error.

**Review by guitargeek — APPROVED** [review]
Thanks again for doing this!

**Comment by egpbos** [comment]
Together with @guitargeek, we found and fixed (in commit abf72d623f3654b20af73b5cdbb4e6afc05e7ba6) a bug in RooNLLVarNew, accidentally proving the usefulness of this PR 😄

**Comment by egpbos** [comment]
The last rebase doesn't change any contents, just made it consistent with the branch in #11940.

Still need to wait for CI.

**Comment by egpbos** [comment]
Ok, roottest has some failures too. I'll see whether I can fix those too.

**Comment by egpbos** [comment]
I fixed roottest, see https://github.com/root-project/roottest/pull/928. I think after that this can be merged.

**Comment by egpbos** [comment]
@phsft-bot build

**Comment by egpbos** [comment]
The roottest fixes were merged, so now CI should pass here.

**Pull request merged**
